pkg/sql/sqlbase/structured.go

```diff
@@ -3699,22 +3699,56 @@ func (desc *Descriptor) GetName() string {
 	}
 }
 
-// Table is a replacement for GetTable() which seeks to ensure that clients
-// which unmarshal Descriptor structs properly set the ModificationTime on
-// tables based on the MVCC timestamp at which the descriptor was read.
-//
-// A linter should ensure that GetTable() is not called.
-func (desc *Descriptor) Table(ts hlc.Timestamp) *TableDescriptor {
-	t := desc.GetTable()
-	if t != nil {
-		t.maybeSetTimeFromMVCCTimestamp(ts)
+// GetVersion returns the Version of the descriptor.
+func (desc *Descriptor) GetVersion() DescriptorVersion {
```

Comment:
Should these panic if the type doesn't match? We want to defend against people adding new descriptors without adding the cases here too.

Reply:
Yeah, done.